command-line/lib/Options/Applicative/Public.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Options.Applicative.Public
+    (
+    -- * Types
+      PublicType (..)
+
+    -- * Applicative Parser
+    , publicOpt
+    ) where
+
+import Prelude
+
+import Options.Applicative
+    ( Parser, flag, long )
+
+
+data PublicType = WithChainCode | WithoutChainCode
```

Comment:
Does a single type and parse function need its own module?

Reply:
I somewhat just complied with the pattern we have for other things. So although I am slightly in favour with your opinion on this this would be outlier and may surprise anyone NEW looking at the code....

Reply:
OK